File: backend/internal/dto/study_plan.go
package dto

type StudyPlanCreateReq struct {
	Title     string `json:"title" binding:"required"`
	SubjectID uint   `json:"subject_id"`
	DailyGoal int    `json:"daily_goal" binding:"required,min=1"`
	StartDate string `json:"start_date" binding:"required"`
	EndDate   string `json:"end_date" binding:"required"`
}

type StudyPlanUpdateReq struct {
	Title     string `json:"title"`
	SubjectID *uint  `json:"subject_id"`
	DailyGoal *int   `json:"daily_goal"`
	StartDate string `json:"start_date"`
	EndDate   string `json:"end_date"`
}

type StudyPlanResp struct {
	ID           uint    `json:"id"`
	SubjectID    uint    `json:"subject_id"`
	Title        string  `json:"title"`
	DailyGoal    int     `json:"daily_goal"`
	StartDate    string  `json:"start_date"`
	EndDate      string  `json:"end_date"`
	Status       int     `json:"status"`
	TodayDone    int64   `json:"today_done"`
	TodayGoal    int     `json:"today_goal"`
	OverallDone  int64   `json:"overall_done"`
	OverallGoal  int     `json:"overall_goal"`
	CompletionPct float64 `json:"completion_pct"`
	RemainingDays int    `json:"remaining_days"`
	CreatedAt    string  `json:"created_at"`
}

File: frontend/src/types/studyPlan.ts
export interface StudyPlanCreateReq {
  title: string
  subject_id?: number
  daily_goal: number
  start_date: string
  end_date: string
}

export interface StudyPlanUpdateReq {
  title?: string
  subject_id?: number
  daily_goal?: number
  start_date?: string
  end_date?: string
}

export interface StudyPlanResp {
  id: number
  subject_id: number
  title: string
  daily_goal: number
  start_date: string
  end_date: string
  status: number
  today_done: number
  today_goal: number
  overall_done: number
  overall_goal: number
  completion_pct: number
  remaining_days: number
  created_at: string
}
